plugins/modules/mso_schema_site_anp.py

```diff
@@ -139,13 +139,9 @@ def main():
 
     mso = MSOModule(module)
 
-    # Get schema_id
-    schema_obj = mso.get_obj('schemas', displayName=schema)
-    if not schema_obj:
-        mso.fail_json(msg="Provided schema '{0}' does not exist".format(schema))
-
-    schema_path = 'schemas/{id}'.format(**schema_obj)
-    schema_id = schema_obj.get('id')
+    # Get schema objects
+    schema_id = mso.lookup_schema(schema)
+    schema_path, schema_obj = mso.query_schema(schema)
```

Comment:
Any reason not to return schema_id as part of query_schema?
This way you are making 2 times a query to the list of schemas where you could make only one.